src/transaction/boot.h

```diff
@@ -175,4 +184,98 @@ COMPOSE_FULL_NAME (char *buf, size_t buf_size, const char *path, const char *nam
       abort ();
     }
 }
+
+/*
+ * boot_remove_useless_path_separator () - Remove useless PATH_SEPARATOR in path string
+ *
+ * return : true or false(in case of fail)
+ *
+ *   path(in): Original path.
+ *   new_path(out): Transformed path.
+ *
+ * Note: This function removes useless PATH_SEPARATOR in path string.
+ *       For example,
+ *       /home3/CUBRID/DB/               -->  /home3/CUBRID/DB
+ *       C:\CUBRID\\\Databases\\         -->  C:\CUBRID\Databases
+ *       \\pooh\user\                    -->  \\pooh\user
+ *
+ *       After transform..
+ *       If new path string is "/" or "\", don't remove the last slash.
+ *       It is survived.
+ */
+inline void
+boot_remove_useless_path_separator (const char *path, char *new_path)
```

Comment:
This function should be guarded with `PATH_MAX`.

Reply:
I know the function is only moved, not newly implemented. :)
I was saying that it is not safe. If input `path` is larger than `PATH_MAX` there is no check to ensure that the function does not continue writing into the destination `new_path` beyond a reasonable `PATH_MAX` length.

Reply:
I understand. But since this function is C-style, there is little the function can do for a real guarantee. E.g. we cannot guarantee that the caller really allocates PATH_MAX size for `new_path`.

I guess all we can do is to make the assertion that `std::strlen (path)` is less than PATH_MAX (although it is late to be caught here).

Reply:
True that.